tests/determinism/test

```diff
@@ -0,0 +1,107 @@
+#!/bin/bash -e
+. "$(dirname "$0")"/../common.sh
+
+# Test that verifies TASTy files can be generated deterministically
+echo "Testing TASTy file generation with Scala 3..."
+
+# Clean any previous builds
+bazel clean
+
+# Build the Scala 3 target, explicitly using Scala 3 toolchain
+echo "Building target with Scala 3 toolchain..."
+bazel build --@rules_scala_annex//rules/scala:scala-toolchain=test_zinc_3 --keep_going --remote_executor= --remote_cache= --disk_cache= :tasty_test_lib 
```

Comment:
Is `--@rules_scala_annex//rules/scala:scala-toolchain=test_zinc_3` necessary? The toolchain is already specified via `scala_toolchain_name`. Also, do we need `--keep_going`?